dlt/destinations/impl/dummy/configuration.py

```diff
@@ -34,6 +34,8 @@ class DummyClientConfiguration(DestinationClientConfiguration):
     """raise terminal exception in job init"""
     fail_transiently_in_init: bool = False
     """raise transient exception in job init"""
+    truncate_tables_on_staging_destination_before_load: bool = True
```

Comment:
my take is to not change dummy at all but write a single test that works on all destinations that support staging destination. see review comment below

Reply:
left both tests -- dummy based and with real destinations.
We need to implement the `should_truncate_table_before_load_on_staging_destination` in the dummy destination anyway.